### ISSUE TITLE
[Bug] std.fs.realpath / std.fs.openFileAbsolute - unable to interact with the filesystem on Linux

### ISSUE THREAD

**Issue by nitrogenez** [ziglang/zig#16935]
### Zig Version

0.11.0

### Steps to Reproduce and Observed Behavior

1. Install Zig 0.11.0 from the official binary package;
2. Write the following code:

```zig
...
var pathbuf: [fs.MAX_PATH_BYTES]u8 = undefined;
const path = try fs.realpath("./main.zig", &pathbuf);
const locales_file = try fs.openFileAbsolute(path, .{});
defer locales_file.close();
...
```

### Expected Behavior

I have expected that code to just open a file in RO mode. But it throws a comptime error, which is following:

```zig
/usr/lib/zig/std/os/linux.zig:734:32: error: unable to evaluate comptime expression
        return syscall3(.open, @intFromPtr(path), flags, perm);
                               ^~~~~~~~~~~~~~~~~
/usr/lib/zig/std/os/linux.zig:734:44: note: operation is runtime due to this operand
        return syscall3(.open, @intFromPtr(path), flags, perm);
                                           ^~~~
/usr/lib/zig/std/os.zig:1492:28: note: called from here
        const rc = open_sym(file_path, flags, perm);
                   ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/usr/lib/zig/std/os.zig:5104:25: note: called from here
        const fd = openZ(pathname, flags, 0) catch |err| switch (err) {
                   ~~~~~^~~~~~~~~~~~~~~~~~~~
/usr/lib/zig/std/os.zig:5091:21: note: called from here
    return realpathZ(&pathname_c, out_buffer);
           ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
src/main.zig:57:37: note: called from here
        const path = try fs.realpath("./main.zig", &pathbuf);
```

**Comment by leecannon** [comment]
For some reason zig has decided that the call to `realpath` is happening in a comptime context.

Can you show the rest of the function?

**Comment by nitrogenez** [comment]
> For some reason zig has decided that the call to `realpath` is happening at comptime.
> 
> Can you show the rest of the function?

This is it. I just wanted to learn how to manipulate files in Zig. And that's the result.

**Comment by nitrogenez** [comment]
> 

That's the problem. I don't want to use file contents at comptime. So, what do I do?

**Comment by nitrogenez** [comment]
I am really sorry for wasting your time, I found the problem. The error message is not really saying anything, but it's all coming from my little knowledge of how Zig works and from my sleep deprivation.

The problem was, I thought that the rest of the function was unrelated, so I didn't include it. But the function was reading file contents and returning them as a `std.ArrayList`. And I tried to make it so file contents will be stored in a `const out`, which was outside of any function body. And I completely forgot that consts initializing outside the function body are initializing at comptime. So that's what was happening.

`const out` initialization was like so:

```zig
...
const out = parse();

pub fn main() !void {...}

pub fn parse() !std.ArrayList([]const u8) {...}
```

So, `out` was initialized at comptime, causing that error. I really do apologise for taking your time. I am closing the issue as complete. The cause was a common human error. Thank you.
